engines/mxnet/jnarator/build.gradle

```diff
@@ -33,3 +35,7 @@ jar {
     duplicatesStrategy = DuplicatesStrategy.INCLUDE
     from configurations.runtimeClasspath.collect { it.isDirectory() ? it : zipTree(it) }
 }
+
+verifyJava.dependsOn generateGrammarSource
```

Comment:
i added the same comment below but can we consolidate all the logic related to verifyJava in one place?

another point, we should use the task configuration avoidance API. this would mean it would look like `dependsOn(tasks.named(generateGrammarSource))`

See more information on task configuration avoidance here: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html

Note: please do not go in and update all the uses of the eager task configuration, that would be a lot of work, we should just ensure that newly added uses are lazy.

Reply:
Hmm, I tried doing this but the way `verifyJava` is defined, it just ends up sprouting a bunch of 'not found' errors in various modules:

`Task with name 'generateTestGrammarSource' not found in project ':api'`

I assume this is because the dependencies are only relevant for a few projects which actually declare the relevant tasks, but `verifyJava` is applied globally.